packages/dds/tree/src/shared-tree-core/sharedTreeCore.ts

```diff
@@ -217,23 +218,28 @@ export class SharedTreeCore<TEditor extends ChangeFamilyEditor, TChange> extends
 	 * @param revision - The revision to associate with the change.
 	 * Defaults to a new, randomly generated, revision if not provided.
 	 */
-	protected applyChange(change: TChange, revision?: RevisionTag): void {
-		const commit = {
+	protected applyChange(change: TChange, revision?: StableId): void {
+		const commit = this.addLocalChangeWithRevision(
 			change,
-			revision: revision ?? mintRevisionTag(),
-			sessionId: this.editManager.localSessionId,
-		};
-		const delta = this.editManager.addLocalChange(commit.revision, change, false);
-		this.transactions.repairStore?.capture(
-			this.changeFamily.intoDelta(change),
-			commit.revision,
+			revision ? revision : mintRevisionTag(),
 		);
 		if (this.transactions.size === 0) {
 			this.submitCommit(commit);
 		}
+	}
+
+	private addLocalChangeWithRevision(change: TChange, revision: StableId): Commit<TChange> {
```

Comment:
```suggestion
	private addLocalChangeWithRevision(change: TChange, revision: RevisionTag): Commit<TChange> {
```

Reply:
and nit: just call it `addLocalChange`. The fact that it requires a revision is clear in the signature, and I don't think the words `WithRevision` give any other helpful information to a caller.